pywbem_mock/_instancewriteprovider.py

```diff
@@ -338,18 +338,13 @@ class for the new instance.
 
         # Parameter types are already checked by WBEMConnection operation
         assert isinstance(namespace, six.string_types)
+        assert isinstance(NewInstance, CIMInstance)
 
-        new_instance = NewInstance
+        new_instance = NewInstance.copy()
```

Comment:
No.
There are two things worthwhile discussing:
* Is the original object modified: In the prior code, there was no copy at all before the path of the original object was modified. The new code shallow-copies the original object before modifying the path of the new object, which leaves the original object unchanged. So that part has improved.
* Which object is put into the repo: That is the same before and after the change, i.e. the original object is offered to `instance_store.create()` which performs a deep copy.